opengrok-indexer/src/main/java/org/opengrok/indexer/web/Util.java

```diff
@@ -844,7 +844,7 @@ public static void readableLine(int num, Writer out, Annotation annotation,
      * @return the mangled path.
      */
     public static String path2uid(String path, String date) {
-        return path.replace('/', '\u0000') + "\u0000" + date;
+        return path.replace(File.separatorChar, '\u0000') + "\u0000" + date;
```

Comment:
how is this going to interact with path sanitization done in populateDocument() ?

Reply:
it makes that sanitization kinda useless, hmm, I think we need to define what to do and how
originally whole opengrok used "/" everywhere - newer versions of windows don't have a problem to recognize /var/opengrok and translate it automatically to C:\var\opengrok
so if we are going to skip File.separator and go back to "/" everywhere it should work - why was it fixed to File.separator? (looking at the changesets I couldn't find the original bugs, resp. the problem in them)

Reply:
why was the path sanitized? did it conflict with lucene somehow, how? /me curious

Reply:
Lucene does not like literal backslashes because it uses them for escaping.

Reply:
As for the sanitization we need to decide where it will be done. If in populateDocument() or in path manipulating methods.